util-core/src/main/scala/com/twitter/io/ByteReader.scala

```diff
@@ -384,6 +398,41 @@ private class ByteReaderImpl(buf: Buf) extends ByteReader {
     ret
   }
 
+  private val unsignedLongMaxValue: BigInt = BigInt("18446744073709551615")
+
+  def readUnsignedLongBE(): BigInt = {
```

Comment:
this makes a lot of intermediate allocations that I think we can probably skip.  how about we use regular longs except for `BigInt(buf.get(pos    ) & 0xff) << 56` and then or them at the end?

also, why do we need to & it with the unsignedLongMaxValue?